### PR TITLE
[SPARK-47307][SQL][FOLLOWUP] Promote spark.sql.legacy.chunkBase64String.enabled from a legacy/internal config to a regular/public one

### PR DESCRIPTION
**Pull request by wForget**
### What changes were proposed in this pull request?

+ Promote spark.sql.legacy.chunkBase64String.enabled from a legacy/internal config to a regular/public one.
+ Add test cases for unbase64

### Why are the changes needed?

Keep the same behavior as before. More details: https://github.com/apache/spark/pull/47303#issuecomment-2237785431

### Does this PR introduce _any_ user-facing change?

yes, revert behavior change introduced in #47303

### How was this patch tested?

existing unit test

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -3526,13 +3526,12 @@ object SQLConf {
     .createWithDefault(false)
 
   val CHUNK_BASE64_STRING_ENABLED = buildConf("spark.sql.legacy.chunkBase64String.enabled")
```
Comment (yaooqinn) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:3528]:
```suggestion
  val CHUNK_BASE64_STRING_ENABLED = buildConf("spark.sql.chunkBase64String.enabled")
```

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
cc @gatorsmile too

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala [hunk 2]
```diff
@@ -520,6 +520,10 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     withSQLConf(SQLConf.CHUNK_BASE64_STRING_ENABLED.key -> "true") {
       checkEvaluation(Base64(Literal(longString.getBytes)), chunkEncoded)
     }
+
+    // check if unbase64 works well for chunked and non-chunked encoded strings
+    checkEvaluation(StringDecode(UnBase64(Literal(encoded)), Literal("utf-8")), longString)
+    checkEvaluation(StringDecode(UnBase64(Literal(chunkEncoded)), Literal("utf-8")), longString)
```
Comment (wForget) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala:526]: I added this case to confirm that the chunkEncoded string can be decoded correctly. cc @cloud-fan

**Review by dongjoon-hyun — COMMENTED** [review]
cc @gatorsmile and @cloud-fan

**Comment by dongjoon-hyun** [comment]
BTW, @wForget , if you change the configuration name in this PR, the PR title and description should be updated accordingly.

Currently, the PR title and description is not matched.

**Comment by dongjoon-hyun** [comment]
In addition, the migration guide too.
```
$ git grep spark.sql.legacy.chunkBase64String.enabled
docs/sql-migration-guide.md:- Since 3.5.2, the `base64` function will return a non-chunked string. To restore the behavior of chunking base64 encoded strings into lines of at most 76 characters, set `spark.sql.legacy.chunkBase64String.enabled` to `true`.
```

**Comment by wForget** [comment]
> BTW, @wForget , if you change the configuration name in this PR, the PR title and description should be updated accordingly.
> 
> Currently, the PR title and description is not matched.

Updated, thank you.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Let's wait for @gatorsmile and @cloud-fan 's review.

**Comment by wForget** [comment]
The test golden files need to be regenerated, I am doing it.

**Comment by dongjoon-hyun** [comment]
Oh, this one? 
```
[info] *** 1 TEST FAILED ***
[error] Failed tests:
[error] 	org.apache.spark.sql.connect.ProtoToParsedPlanTestSuite
```

**Comment by wForget** [comment]
> Oh, this one?

yeah

**Comment by yaooqinn** [comment]
Merged to master.

Hi @wForget, please help backport this to 3.5. Thank you

**Comment by cloud-fan** [comment]
It's good to keep the old behavior by default, but I'd like to understand the difference. Which one is more commonly supported in the ecosystem (decoding)? The chunked one or the unchunked one?

**Comment by yaooqinn** [comment]
Unchunking is the behavior before 3.2(inclusive). Chunking was the behavior introduced accidentally in 3.3. I suggested @wForget in #47303 to make chunking as legacy behavior and unchunking as default, considering that unchunking is more reasonable here and users may not upgrade so eagerly, while chunking is mostly for editors who can not handle long lines. For the unbase64 part, both are valid for decoding.

**Comment by wForget** [comment]
Hive, Trino, and Clickhouse all seem to be unchunked.

Hive: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFBase64.java

Trino: https://github.com/trinodb/trino/blob/4acd387d3e3dd739a8b2ed00fb48f9bbb00c5ab0/core/trino-main/src/main/java/io/trino/operator/scalar/VarbinaryFunctions.java#L61-L64

Clickhouse: https://github.com/ClickHouse/ClickHouse/blob/a07707c56e5c5ad75b19a0e6f75ec02f0ab6c481/src/Common/Base64.cpp#L18

**Pull request closed without merging**